services/supervisor/tests/supervisor/utils.rs

```diff
@@ -0,0 +1,176 @@
+    SERVICE_NAME as CONFIG_SERVICE_NAME,
+};
+use exonum_supervisor::{ConfigPropose, ConfigVote, Schema, Supervisor};
+
+pub const SECOND_SERVICE_ID: InstanceId = 119;
+pub const SECOND_SERVICE_NAME: &str = "change-service";
+
+pub fn config_propose_entry(testkit: &TestKit) -> Option<ConfigPropose> {
+    let snapshot = testkit.snapshot();
+    Schema::new(SUPERVISOR_INSTANCE_NAME, &snapshot)
+        .config_propose_with_hash_entry()
+        .get()
+        .map(|entry| entry.config_propose)
+}
+
+pub fn sign_config_propose_transaction(
+    testkit: &TestKit,
+    config: ConfigPropose,
+    initiator_id: ValidatorId,
+) -> Verified<AnyTx> {
+    let keys = &testkit.validator(initiator_id).service_keypair();
+    config.sign(SUPERVISOR_INSTANCE_ID, keys.0, &keys.1)
+}
+
+pub fn build_confirmation_transactions(
+    testkit: &TestKit,
+    proposal_hash: Hash,
+    initiator_id: ValidatorId,
+) -> Vec<Verified<AnyTx>> {
+    testkit
+        .network()
+        .validators()
+        .iter()
+        .filter(|validator| validator.validator_id() != Some(initiator_id))
+        .map(|validator| {
+            let keys = validator.service_keypair();
+            ConfigVote {
+                propose_hash: proposal_hash,
+            }
+            .sign(SUPERVISOR_INSTANCE_ID, keys.0, &keys.1)
+        })
+        .collect()
+}
+
+pub struct ConfigProposeConstructor {
+    config_propose: ConfigPropose,
+}
+
+impl ConfigProposeConstructor {
+    pub fn new(cfg_change_height: Height) -> Self {
+        ConfigProposeConstructor {
+            config_propose: ConfigPropose {
+                actual_from: cfg_change_height,
+                changes: vec![],
+            },
+        }
+    }
+
+    pub fn extend_consensus_config_propose(mut self, consensus_config: ConsensusConfig) -> Self {
+        self.config_propose
+            .changes
+            .push(ConfigChange::Consensus(consensus_config));
+        self
+    }
+
+    pub fn extend_service_config_propose(mut self, param: String) -> Self {
+        self.config_propose
+            .changes
+            .push(ConfigChange::Service(ServiceConfig {
+                instance_id: CONFIG_SERVICE_ID,
+                params: param.into_bytes(),
+            }));
+        self
+    }
+
+    pub fn extend_second_service_config_propose(mut self, param: String) -> Self {
+        self.config_propose
+            .changes
+            .push(ConfigChange::Service(ServiceConfig {
+                instance_id: SECOND_SERVICE_ID,
+                params: param.into_bytes(),
+            }));
+        self
+    }
+
+    pub fn config_propose(&self) -> ConfigPropose {
+        self.config_propose.clone()
+    }
+}
+
+pub fn consensus_config_propose_first_variant(testkit: &TestKit) -> ConsensusConfig {
+    let mut cfg = testkit.consensus_config();
+    // Change any config field.
+    // For test purpose ut doesn't matter what exactly filed will be changed.
+    cfg.min_propose_timeout = 20;
```

Comment:
Nit: If `testkit.consensus_config()` will someday be returning config with `min_propose_timeout` == 20, some tests will eventually become incorrect. I'd suggest `cfg.min_propose_timeout += 1` for first variant and `+= 2` for second variant.